### PR TITLE
[OpenMP] [Flang] [MLIR] Allow dist_schedule(static) to be default behavior in OMP DISTRIBUTE directive

### PR DESCRIPTION
**Pull request by raghavendhra**
According to the OpenMP spec, the default schedule for `distribute` is implementation-defined. We effectively implement a static schedule by default, so we simply parse and ignore the `dist_schedule` clause instead of flagging it as unimplemented error message. 

After this change,

    !$omp target teams distribute parallel do dist_schedule(static) -> No longer issues unimplemented error message for dist_schedule static only case
    
    !$omp target teams distribute parallel do dist_schedule(static, chunk_size) -> Issues unimplemented error message for dist_schedule chunk_size

### CODE REVIEW

File: mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp [hunk 1]
```diff
@@ -169,8 +169,8 @@ static LogicalResult checkImplementationStatus(Operation &op) {
       result = todo("device");
   };
   auto checkDistSchedule = [&todo](auto op, LogicalResult &result) {
-    if (op.getDistScheduleStatic() || op.getDistScheduleChunkSize())
-      result = todo("dist_schedule");
+    if (op.getDistScheduleChunkSize())
+      result = todo("dist_schedule chunk_size");
```
Comment (mjklemm) [inline mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp:173]:
```suggestion
      result = todo("dist_schedule with chunk_size");
```

File: mlir/test/Target/LLVMIR/openmp-todo.mlir [hunk 2]
```diff
@@ -80,7 +80,7 @@ llvm.func @distribute_allocate(%lb : i32, %ub : i32, %step : i32, %x : !llvm.ptr
 // -----
 
 llvm.func @distribute_dist_schedule(%lb : i32, %ub : i32, %step : i32, %x : i32) {
-  // expected-error@below {{not yet implemented: Unhandled clause dist_schedule in omp.distribute operation}}
+  // expected-error@below {{not yet implemented: Unhandled clause dist_schedule chunk_size in omp.distribute operation}}
```
Comment (mjklemm) [inline mlir/test/Target/LLVMIR/openmp-todo.mlir:83]:
```suggestion
  // expected-error@below {{not yet implemented: Unhandled clause dist_schedule with chunk_size in omp.distribute operation}}
```

**Review by mjklemm — APPROVED** [review]
LGTM

**Review by skatrak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**